src/types/routeFlat.ts

```diff
@@ -1,5 +1,9 @@
+import { Param } from '@/types/params'
+import { Path } from '@/utilities'
+
 export type RouteFlat = {
   name: string,
-  path: string,
+  path: string | Path,
```

Comment:
Curious your thoughts of making this a union. At this point I assumed we'd always be dealing with a string since the params are a separate property.